zerver/lib/user_groups.py

```diff
@@ -326,6 +346,14 @@ def update_or_create_user_group_for_setting(
             _("Invalid user group ID: {group_id}").format(group_id=group_ids_not_found[0])
         )
 
+    for subgroup in potential_subgroups:
+        # At this time, we only do a check on the realm ID of the subgroup and
+        # whether the group is deactivated or not. Realm ID error would be caught
+        # above and in case the user group is deactivated the error will be raised
+        # in has_user_group_access itself, so there is no coverage here.
+        if not has_user_group_access(subgroup, user_profile, for_read=False, as_subgroup=True):
```

Comment:
We could have filter out the deactivated groups above in the `NamedUserGroup` query but this seemed better to give a meaningful message mentioning that the group is deactivated.